lm_eval/tasks/ja/jaqket_v2.py

```diff
@@ -489,6 +489,28 @@ class JAQKETV2WithLlama2(JAQKETV2WithJAAlpacaPrompt):
     DESCRIPTION = f"<s>[INST] <<SYS>>\n{SYSTEM_PROMPT}\n<</SYS>>\n\n"
     FEWSHOT_SEP = " </s><s>[INST] "
 
+    def preprocess_ctx(self, ctx, max_length):
+        # if ctx fits in max length, return
+        if len(self.tokenizer.encode(ctx)) <= max_length:
+            return ctx
+
+        # if ctx is too long, split on a tag that separates each example
+        _, remainder = ctx.split(self.DESCRIPTION, 1)
+        ctxs = remainder.split(self.FEWSHOT_SEP)
```

Comment:
The behavior of this is correct, but I find using `split` to remove the description extremely weird and confusing. Would this not be equivalent?

```
ctxs = ctx[len(self.DESCRIPTION):].split(self.FEWSHOT_SEP)
```

Reply:
Changed based on your comment https://github.com/Stability-AI/lm-evaluation-harness/pull/105#issuecomment-1761075681

So, this line is gone now.